stan/math/prim/mat/meta/append_return_type.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef STAN_MATH_PRIM_MAT_META_APPEND_RETURN_TYPE_HPP
+#define STAN_MATH_PRIM_MAT_META_APPEND_RETURN_TYPE_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <vector>
+
+namespace stan {
+  namespace math {
+    /**
+     * This template metaprogram defines the return type for appending
+     * containers of the specified template parameter types.
+     *
+     * This base implementation assumes the template parameters are not
+     * container types and calculates their return type using
+     * <code>return_type</code>.
+     *
+     * Specializations define behavior for other basic types and for containers.
+     *
+     * @tparam T1 First type to be promoted
+     * @tparam T2 Second type to be promoted
+     */
+    template <typename T1, typename T2>
+    struct append_return_type {
+      typedef typename return_type<T1, T2>::type type;
+    };
+
+    /**
+     * If both types are ints, the type member is an int (this is different
```

Comment:
You should still stay what this does.  

You don't need to doc differences with other functions---there's just no end to that.  You can doc the previous function to indicate it'll deal with all the scalar types other than the `(int, int)` pair.